configs/myconfigs/sampling/Sampling.py

```diff
@@ -0,0 +1,361 @@
+    from random import randint, sample, seed
+    global sample_insts
+    seed()
+
+    warmup_time = opts.warmup_time
+    detailed_warmup_time = opts.detailed_warmup_time
+    detailed_time = opts.detailed_time
+
+    # We probably want to updated the max instructions by subtracting the
+    # insts that we are going to execute in the last detailed sim.
+    sample_secs = toLatency(warmup_time) + toLatency(detailed_warmup_time) \
+                    + toLatency(detailed_time)
+    # assume 1 IPC per CPU
+    instructions -= int(sample_secs * system.clk_domain.clock[0].frequency *
+                        len(system.cpu) * 1)
+
+    # Get the instruction numbers that we want to exit at for our sample
+    # points. It needs to be sorted, too.
+    sample_insts = sample(xrange(start_insts,
+                                 start_insts + instructions),
+                          samples)
+    sample_insts.sort()
+
+
+
+def dumpSamples(outfile="random.samples"):
+    """ Dumps the sample points in a file so that they can
+        be used accross multiple parallel jobs
+
+        @param outfile name of the sample dump file
+    """
+    import pickle
+    rfilename = "%s/%s" % (m5.options.outdir, outfile)
+
+    print "Dumping random sample points at :", rfilename
+    rfile = open(rfilename , "w")
+    pickle.dump(sample_insts, rfile)
+    rfile.close()
+
+
+
+def loadSamples(samplefile):
+    """ Loads sample points from a file dumped previously
+        using dumpSamples().
+
+        @param samplefile name of the sample dump file
+    """
+    import pickle
+    global sample_insts
+
+    print "Loading random samples : ", samplefile
+    if not os.path.isfile(samplefile):
+        print >> sys.stderr, "Error could not find file", samplefile
+        sys.exit(1)
+
+    rfile = open(samplefile, "r")
+    sample_insts = pickle.load(rfile)
+    rfile.close()
+
+
+def makeSampleDir(sampleno):
+    """ Make directory for a sample
+        Returns the name of the directory
+        @param sampleno sample id number """
+    insts = sample_insts[sampleno]
+    parent = m5.options.outdir
+    sampledir = '%(parent)s/'%{'parent':parent} + str(insts)
+    if not os.path.isdir(sampledir):
+        os.mkdir(sampledir)
+    return sampledir
+
+
+def forwardToSample(system, sample):
+    """ Fast forward to a particular sample number.
+
+        @param system the system we are running
+        @param sample to forward to
+    """
+    global _sample
+    insts = sample_insts[sample]
+    insts_past = fastforward(system, insts - system.totalInsts())
+    if insts_past/insts > 0.01:
+        print "WARNING: Went past the goal instructions by too much!"
+        print "Goal: %d, Actual: %d" % (insts, system.totalInsts())
+    _sample = sample
+
+
+
+def checkSystem(system):
+    """ Do some basic validation of the system such as does it
+        have the required objects, uses a KVM cpu etc.
+        This also adds some helper functions used by the Sampling
```

Comment:
Wow, this is tricky! Specifically say that it modifies the system instance with helper functions. Name the functions that it's adding as well.

I like the cleverness, but it's tricky, so let's be really clear as to what's going on.